.claude-plugin/plugins/limacharlie-skills/skills/data-source-connector/SKILL.md

```diff
@@ -0,0 +1,412 @@
+2. LimaCharlie adapter docs: https://docs.limacharlie.io/docs/adapter-usage
+3. Specific adapter configuration for [ADAPTER_TYPE]
+
+Return:
+- Required credentials/prerequisites for [DATA_SOURCE]
+- Complete configuration YAML with all required parameters
+- CLI command examples for running the adapter
+- Common issues and troubleshooting tips
+- Platform type to use (json, aws, azure_monitor, etc.)
+"
+```
+
+**IMPORTANT**: Always spawn this sub-agent BEFORE providing detailed configuration instructions. The USP adapters repository is the source of truth for adapter configuration.
+
+---
+
+## CRITICAL: Organization IDs, Sensor IDs, and UUIDs
+
+**NEVER fabricate, guess, or invent OIDs, SIDs, or any UUID values.**
+
+LimaCharlie uses several types of identifiers that you must handle correctly:
+
+| Identifier | Format | Example | Description |
+|------------|--------|---------|-------------|
+| **OID** (Organization ID) | UUID | Example: `95de69c2-5a29-4378-aab2-efdcfaa044f7` | Unique identifier for a LimaCharlie organization |
+| **SID** (Sensor ID) | UUID | Example: `a4b3c2d1-1234-5678-abcd-123456789abc` | Unique identifier for a sensor/endpoint |
+| **IID** (Installation Key ID) | UUID | Example: `a2f7d425-a256-4efe-880c-5b8c1f530c11` | Identifier for an installation key |
+
+**These are NOT valid identifiers:**
+- Organization names like `my-company` or `prod-environment`
+- Hostnames like `web-server-01` or `gcp-lc-sales-audit`
+- Project names like `lc-sales`
+- Arbitrary strings or labels
+
+### Critical Rules
+
+**NEVER**:
+- Fabricate or invent OIDs, SIDs, or any UUID values
+- Use placeholder UUIDs like `00000000-0000-0000-0000-000000000000`
+- Guess at organization or sensor identifiers
+- Assume a UUID format value without verification
+
+**ALWAYS**:
+- **Ask the user** for their Organization ID (OID) if not provided
+- **Use MCP tools** to retrieve valid identifiers:
+  - `mcp__limacharlie__list_user_orgs` - Get accessible organizations
+  - `mcp__limacharlie__list_sensors` - Get sensors in an organization
+  - `mcp__limacharlie__search_hosts` - Find sensors by hostname
+  - `mcp__limacharlie__list_installation_keys` - Get installation keys
+- **Validate** that identifiers are in proper UUID format before use
+- **Show users** where to find their OID in the LimaCharlie web UI (Access Management → REST API)
+
+### Where Users Find Their OID
+
+Direct users to find their Organization ID:
+1. Log in to https://app.limacharlie.io
+2. Navigate to **Access Management** → **REST API**
+3. The OID is displayed on this page
+
+### Example: Asking for OID
+
+When a user wants to connect a data source but hasn't provided their OID:
+
+```
+"To configure this adapter, I'll need your LimaCharlie Organization ID (OID).
+You can find this in the LimaCharlie web app under Organization Settings.
+What's your OID?"
+```
+
+---
+
+## Available USP Adapters Catalog
+
+### Cloud Storage & Messaging (7 adapters)
```

Comment:
I suggest removing these and making it auto-discove the adapter types from the usp-adapters repo.

Reply:
I removed all static references and emphasized the use of documentation for any data that could by dynamic.